actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -7,7 +7,12 @@ def send_action(method, *args)
     end
 
     def default_render(*args)
-      render(*args)
+      if template_exists?(action_name.to_s, _prefixes, variants: request.variant)
```

Comment:
I don't think it is cached although I might not have gone deep enough: https://github.com/rails/rails/blob/08754f12e65a9ec79633a605e986d0f1ffa4b251/actionview/lib/action_view/path_set.rb

I like it being a part of the public interface, but I see your point about double checking (albeit it is exceptionally). This case would come up in the situation that your controller has an action w/o a corresponding method to call, but there is a template matching that action.